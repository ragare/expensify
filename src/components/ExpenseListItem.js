import React from 'react'
import moment from 'moment'
import { Link } from 'react-router-dom'
import numeral from 'numeral'

const ExpenseListItem = ({ id, description, amount, createdAt }) => (
    <div>
        <Link to={`/edit/${id}`}>
            <h1>{description}</h1>
        </Link>
        <p>
            {numeral(amount / 100).format('$0,0.00')}
            -
        {moment(createdAt).format('MMM Do, YYYY')}
        </p>
    </div>
)

export default ExpenseListItem