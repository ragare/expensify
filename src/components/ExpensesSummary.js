import React from 'react'
import { connect } from 'react-redux'
import { Link } from 'react-router-dom'
import selectExpenses from '../selectors/expenses'
import selectExpensesTotal from '../selectors/expenses-total'
import numeral from 'numeral'

export const ExpensesSummary = (props) => (
    <div className="page-header">
        <div className="content-container">
            <h1 className="page-header__title">
                Viewing <span>{props.expenseCount}</span> totalling <span>{numeral(props.expensesTotal / 100).format('$0,0.00')}</span>
            </h1>
            <div className="page_header__actions">
                <Link className="button" to="/create">Add Expense</Link>
            </div>
        </div>
    </div>
)


const mapStateToProps = (state) => {
    const expenses = selectExpenses(state.expenses, state.filters)
    return {
        expenseCount: expenses.length,
        expensesTotal: selectExpensesTotal(expenses)
    };
}

export default connect(mapStateToProps)(ExpensesSummary);
