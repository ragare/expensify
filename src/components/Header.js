import React from 'react'
import { NavLink } from 'react-router-dom'
import { connect } from 'react-redux'
import { startLogOut } from '../actions/auth'

export const Header = ({ startLogOut }) => (
    <header>
        <h1>Expensify</h1>
        <NavLink to="/" activeClassName="is-active" exact={true}>Home</NavLink>
        <NavLink to="/create" activeClassName="is-active">Create</NavLink>
        <button onClick={startLogOut}>Logout</button>
    </header>
)

const mapDispatchToProps = (dispatch) => ({
    startLogOut: () => dispatch(startLogOut())
})

export default connect(undefined, mapDispatchToProps)(Header)