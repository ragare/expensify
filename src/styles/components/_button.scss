.button {
    background-color: $button-blue;
    border: none;
    color: white;
    font-size: 1.8rem;
    font-weight: 300;
    padding: $s-size;
}