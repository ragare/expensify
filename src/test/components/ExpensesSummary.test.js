import React from 'react'
import { shallow } from 'enzyme'
import { ExpensesSummary } from '../../components/ExpensesSummary'

test('should sow a summary with 1 and a total of $99', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={9900} />)
    expect(wrapper).toMatchSnapshot()
})

test('should sow a summary with 2 and a total of $99', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={2} expensesTotal={9900} />)
    expect(wrapper).toMatchSnapshot()
})