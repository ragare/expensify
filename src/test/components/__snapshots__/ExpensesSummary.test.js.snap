// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should sow a summary with 1 and a total of $99 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        1
      </span>
       totalling 
      <span>
        $99.00
      </span>
    </h1>
    <div
      className="page_header__actions"
    >
      <Link
        className="button"
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should sow a summary with 2 and a total of $99 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        2
      </span>
       totalling 
      <span>
        $99.00
      </span>
    </h1>
    <div
      className="page_header__actions"
    >
      <Link
        className="button"
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
